0000-experiment-dataframe.md

```diff
@@ -0,0 +1,320 @@
+
+#### User3
+
+User wants to easily perform custom analysis on outcomes of experiments. For example, the [probabilistic error cancellation](https://arxiv.org/abs/2201.09866) first estimates fidelities of various Pauli channels with RB family experiment, then an experimentalist will develop a noise model tailored to a quantum circuit with these fidelities.
+
+_Pain points_:
+
+Currently analysis results are retuned as a list of `AnalysisResult` and they are not well organized. The experimentalist needs to iterate over all list items and check the entry name and extra metadata to filter the result of interest.
+
+_Solutions_:
+
+This user wants to use data frame to organize the analysis results so it is easy to work with.
+
+
+## Design Proposal
+
+In this proposal, we are going to replace the data format of analysis results, based off of our users survey.
+
+### A. Experiment data
+
+We propose to replace `ExperimentData._analysis_results` data format from `ThreadSafeOrderedDict` to the thread safe wrapper of the pandas data frame. This data frame may consists of following columns:
+
+- name (str): Same as `AnalysisResultData.result_type`. This name will appear in the result database.
+- value (Any): Same as `AnalysisResultData.result_data["_value"]`. Any value that QE analysis will generate. This value must be serializable through the `ExperimentEncoder`.
+- chisq (float): Same as `AnalysisResultData.chisq`. Chi-squared value of the entry. Probably this can be moved to extra dict.
+- quality (str): Same as `AnalysisResultData.quality`. Systematically evaluated quality of the fitting to yield this data.
+- device_components (List[DeviceComponent]): Same as `AnalysisResultData.device_components`. Usually `.physical_qubits` of the experiment.
+- extra (Dict): Same as `AnalysisResultData.result_data["_extra"]`. Free-form dictionary for analysis metadata.
+- experiment_id (str): Same as `AnalysisResultData.experiment_id`. The ID of the associated experiment.
+- verified (bool): Same as `AnalysisResultData.verified`. A boolean value to indicate if this entry is checked by human experimentalist. This can be removed since this is rarely used.
+- tags (List[str]): Same as `AnalysisResultData.tags`. An arbitrary strings to help filtering this entry.
+- source (Dict): Same as `AnalysisResultData.result_data["_source"]`. Version information of QE that was used to generate this entry.
+- created_in_db (bool): Same as `AnalysisResultData._created_in_db`. A boolean value to indicate wether this is loaded from the database or just created locally.
+- result_id (str): Same as `AnalysisResultData.result_id`. A unique ID of this entry.
+- backend (str): Same as `AnalysisResultData.backend_name`. Name of the associated backend.
+- creation_datetime (datetime): Same as `AnalysisResultData.creation_datetime`. A date time at which the experiment data received the job results.
```

Comment:
It might also be good to keep these field names short, since they will be rows in pandas (eg `created` instead of `creation_datetime`)

Reply:
`created` sounds like a boolean value. I renamed `device_components` -> `components`, and `creation_datetime` -> `date`. Note that creation date time value (date time object repr) is rather lengthy, and this change doesn't shorten the widths of the cells.

Reply:
Done in [8454c5e](https://github.com/Qiskit/rfcs/pull/28/commits/8454c5eeb0a4ad360ef7966b56c5e67e1471b369)